ibis/backends/duckdb/tests/test_geospatial.py

```diff
@@ -347,3 +347,14 @@ def test_cast_wkb_to_geo(con):
     geo_expr = t.geometry.cast("geometry")
     assert geo_expr.type().is_geospatial()
     assert isinstance(con.execute(geo_expr), gpd.GeoSeries)
+
+
+def test_load_spatial_casting(ext_dir):
```

Comment:
If you're going to use a custom extension directory you need to produce a new one specific to this test call.

As is, `ext_dir` is a session-level fixture, which means that if another test runs before this one that happens to install the geospatial extension then you're not testing the thing that needs to be tested (or its success is implicitly depending on the fact that another test ran before it, and that'll lead to a flaky test), which is that geospatial operations load the extension.

Reply:
I want to be able have a new connection that doesn't have the extension loaded. I was following what we did here 
https://github.com/ibis-project/ibis/blob/fead40be6a25f24672348fdefa8b35d0e7469f80/ibis/backends/duckdb/tests/test_geospatial.py#L322-L334

Which sounds like it's also being missed used here too? I can scope it to "function", though I'm not sure that's the path we want.

Reply:
> Which sounds like it's also being missed used here too?

Probably 😅 

If you change the fixture decorator from

```python
@pytest.fixture(scope="session")
```

to 

```python
@pytest.fixture
```

that should be sufficient (no `scope="function"` necessary).